Code/GraphMol/MolDraw2D/MolDraw2D.h

```diff
@@ -79,12 +91,26 @@ class MolDraw2D {
       const std::map<int, double> *highlight_radii = NULL, int confId = -1);
 
   virtual void drawMolecule(
+      const ROMol &mol, const std::string &legend,
+      const std::vector<int> *highlight_atoms = NULL,
+      const std::map<int, DrawColour> *highlight_map = NULL,
+      const std::map<int, double> *highlight_radii = NULL, int confId = -1);
+
+  virtual void drawMolecule(
```

Comment:
this is quite a lot of additional overloads for this API just to support legends. Is the rationale for doing this (as opposed to appending the legend parameter in the existing function) to not break backwards compatibility while still having a non-crappy signature, since I assumes the args are listed in order of most likely used.